src/main/java/com/dawn/jat/illuminati/post/controller/PostController.java

```diff
@@ -32,23 +32,25 @@
      */
     @GetMapping(value = "/{slug}")
     public ResponseEntity<Object> getPostBySlug(@PathVariable("slug") String slug) {
-        Optional<PostEntity> postEntity = postService.findBySlug(slug);
+        PostDto postDto = postService.getPost(slug);
 
-        if (!postEntity.isPresent()) {
+        if (Objects.isNull(postDto)) {
             throw new PostNotFoundException("Cannot find post");
         }
 
-        SuccessResponse resp = new SuccessResponse(postEntity);
+        SuccessResponse resp = new SuccessResponse(postDto);
+
         return new ResponseEntity<>(resp, HttpStatus.OK);
     }
 
     /**
      * Gets list of Post Summary.
+     *
      * @throws PostSummaryNotFoundException if Post Summary has No Content
      */
     @GetMapping(value = "/all-post/summary")
     public ResponseEntity<Object> getAllPostSummary() {
-        List<PostSummaryEntity> allPost = postService.findPostSummary();
+        List<PostSummaryEntity> allPost = postService.getPostList();
```

Comment:
the method name is not consistent. `getPostList` don't make sense when we use PostSummaryEntity. May be we can use `getPostSummaries()`.

Reply:
Updated